client/protocols/analytics.py

```diff
@@ -106,13 +110,14 @@ def log_run(self, messages):
             history['question'] = questions
 
         # Update attempt and correctness counts for the graded questions
-        for question in questions:
+        for question in history['question']:
             detail = history['questions']
```

Comment:
What's the difference between `history['question']` and `history['questions']`? And `history['question']` is a list even though it's singular?